### PR TITLE
Update @wordpress devDependecies

### PR DESCRIPTION
**Pull request by gigitux**
This PR upgrade all the devDependecies with @wordpress/* prefix. The idea is to upgrade the devDependecies with the minimum version of WordPress that WooCommerce Blocks supports.

This update doesn't introduce any risk since all the dependencies are extracted from the bundle using `@wordpress/dependency-extraction-webpack-plugin`, as they are included in WordPress.

By doing so, we can ensure that our code has suitable types and IDE auto-completions (when the dependence shipped them). Additionally, this approach helps prevent compatibility issues from using methods only available on the newer versions (if the package is typed, TS should say that the function doesn’t exist).

Since WP 6.3 is near being released, I think it makes sense to stick them to 6.2.

Furthermore, I created a JS script to update them easily. From the root of the project, you can run:

```
node bin/update-wordpress-dependency.js $wordpress-version
```

P2 post: pdToLP-zq-p2

### CODE REVIEW

**Comment by gigitux** [comment]
It seems that with the last changes, we have some issues with the `peerDependencies`. I'm going to move this in draft and work on in in the next weeks

File: package.json [hunk 1]
```diff
@@ -150,26 +150,26 @@
...
-		"@wordpress/prettier-config": "1.4.0",
-		"@wordpress/scripts": "22.3.0",
+		"@wordpress/e2e-tests": "wp-6.2",
+		"@wordpress/element": "wp-6.2",
+		"@wordpress/env": "wp-6.2",
+		"@wordpress/html-entities": "wp-6.2",
+		"@wordpress/i18n": "wp-6.2",
+		"@wordpress/is-shallow-equal": "wp-6.2",
+		"@wordpress/prettier-config": "wp-6.2",
+		"@wordpress/scripts": "wp-6.2",
```
Comment (dinhtungdu) [inline package.json:172]: TIL about these `wp-*` tags. Interesting!

**Review by dinhtungdu — COMMENTED** [review]
@gigitux Interesting! It's a good way to keep our dependencies up-to-date.

At this point I don't see any issue with this except updating dependencies now take more effort because we're updating all `wordpress` packages. But I see it as the trade off to the benefit we gain.

Let's update the PR with `trunk` and continue testing.

**Comment by gigitux** [comment]
I'm closing this PR, given that it is no longer valid.

**Pull request closed without merging**